js/binance.js

```diff
@@ -979,6 +989,110 @@ module.exports = class binance extends Exchange {
+        //                                                                amount: "0.8012",
+        //                                                           operateTime: "2018-10-07 17:56:07",
+        //                                                      transferedAmount: "0.00628141",
+        //                                                             fromAsset: "ADA"                  } ],
+        //                                 operate_time: "2018-10-07 17:56:06"                                } ] } }
+        let rows = response['results']['rows'];
+        let data = [];
+        for (let i = 0; i < rows.length; i++) {
+            let logs = rows[i]['logs'];
+            for (let j = 0; j < logs.length; j++) {
+                logs[j]['isDustTrade'] = true;
+                data.push (logs[j]);
+            }
+        }
+        const trades = this.parseTrades (data, undefined, since, limit);
+        return this.filterBySinceLimit (trades, since, limit);
+    }
+
+    parseDustTrade (trade, market = undefined) {
+        // {              tranId:  2701371634,
+        //   serviceChargeAmount: "0.00012819",
+        //                   uid: "35103861",
+        //                amount: "0.8012",
+        //           operateTime: "2018-10-07 17:56:07",
+        //      transferedAmount: "0.00628141",
+        //             fromAsset: "ADA"                  },
+        let order = this.safeString (trade, 'tranId');
+        let time = this.safeString (trade, 'operateTime');
+        let timestamp = this.parse8601 (time.replace (' ', 'T') + 'Z');
+        let datetime = this.iso8601 (timestamp);
+        let tradedCurrency = this.safeCurrencyCode (trade, 'fromAsset');
+        let earnedCurrency = this.currency ('BNB')['code'];
+        let tradedCurrencyIsQuote = false;
+        for (let i = 0; i < this.options.quoteCurrencyIds.length; i++) {
+            let asset = this.options.quoteCurrencyIds[i];
+            if (tradedCurrency === this.currency (asset)['code']) {
+                tradedCurrencyIsQuote = true;
+                break;
+            }
+        }
+        //
+        // Warning
+        // Binance dust trade `fee` is already excluded from the `BNB` earning reported in the `Dust Log`.
+        // So the parser should either set the `fee.cost` to `0` or add it on top of the earned
+        // BNB `amount` (or `cost` depending on the trade `side`). The second of the above options
+        // is much more illustrative and therefore preferable.
+        //
+        let fee = {
+            'currency': earnedCurrency,
+            'cost': this.safeFloat (trade, 'serviceChargeAmount'),
+        };
+        let symbol = undefined;
+        let amount = undefined;
+        let cost = undefined;
+        let side = undefined;
+        if (tradedCurrencyIsQuote) {
+            symbol = earnedCurrency + '/' + tradedCurrency;
+            amount = this.safeFloat (trade, 'transferedAmount') + fee['cost'];
```

Comment:
We have to use `this.sum` here, because `+` is transpiled to string-concatenation `.` in PHP, instead of numeric addition `+`: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings